src/Core/Domain/Title/QueryResult/EditableTitle.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Title\QueryResult;
+
+use PrestaShop\PrestaShop\Core\Domain\Title\ValueObject\Gender;
+use PrestaShop\PrestaShop\Core\Domain\Title\ValueObject\TitleId;
+
+/**
+ * Transfers state data for editing
+ */
+class EditableTitle
+{
+    /**
+     * @var TitleId
+     */
+    protected $titleId;
+
+    /**
+     * @var array<string>
+     */
+    protected $localizedNames;
+
+    /**
+     * @var Gender
+     */
+    protected $gender;
+
+    /**
+     * @param TitleId $titleId
+     * @param array<string> $localizedNames
+     * @param Gender $gender
+     */
+    public function __construct(
+        TitleId $titleId,
+        array $localizedNames,
+        Gender $gender
```

Comment:
```suggestion
        int $titleId,
        array $localizedNames,
        int $gender
```

Reply:
Sorry, but in our convention, we don't use VOs in query results but only scalar values (except for DecimalNumber). I don't remember exactly all the reasons but amongst them the idea is that in commands VOs are used partially to prevent invalid inputs (the input in command should be scalar though), but in query result though since the model is supposed to be consistent there can be no invalid values.

And also having mostly scalar values should ease the serialization.

Reply:
Fixed